.github/workflows/release.yml

```diff
@@ -62,10 +70,24 @@ jobs:
             substituters = http://cache.nixos.org https://hydra.iohk.io
             trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=
 
-      - name: Upload release.nix
-        uses: runtimeverification/upload-release.nix@v1.1
+      - name: 'Install Cachix'
+        if: ${{ !startsWith(matrix.os, 'self') }}
+        uses: cachix/cachix-action@v12
+
+      - name: 'Publish K to k-framework-binary cache'
+        uses: workflow/nix-shell-action@v3
+        env:
+          CACHIX_AUTH_TOKEN: '${{ secrets.CACHIX_PRIVATE_KFB_TOKEN }}'
+          GC_DONT_GC: '1'
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
+          packages: jq
+          script: |
+            export PATH="$(nix build github:runtimeverification/kup --no-link --json | jq -r '.[].outputs | to_entries[].value')/bin:$PATH"
+            kup publish k-framework-binary .#k --keep-days 180
+            kup publish k-framework-binary .#k.openssl --keep-days 180
+            kup publish k-framework-binary .#k.procps --keep-days 180
+            kup publish k-framework-binary .#k.openssl.procps --keep-days 180
+            kup publish k-framework-binary .#k.procps.openssl --keep-days 180
```

Comment:
This seems like a possible source of difficulty in the future; is there a reason these two permutations are needed? Shouldn't we normalise the list of extra libraries somehow?

Reply:
We need both unless we normalise `kup install k.procps.openssl` into `kup install k.openssl.procps` by sorting. However, `#k.procps.openssl` and `#k.openssl.procps` evaluate to the same derivation anyway so we basically just tell cachix to pin both keys to the same package

Reply:
note that in a different package these could potentially be non-commutative

Reply:
Sure, that's all I wanted to check - I wasn't sure what the commutative properties were like here!